src/main/java/com/google/devtools/build/lib/rules/cpp/CppHelper.java

```diff
@@ -990,9 +990,24 @@ public static CcDebugInfoContext mergeCcDebugInfoContexts(
         .collect(ImmutableList.toImmutableList());
   }
 
+  private static final Label NOOP_GREP_INCLUDES =
+      Label.parseAbsoluteUnchecked("@bazel_tools//tools/cpp:grep-includes");
+
   public static Artifact getGrepIncludes(RuleContext ruleContext) {
```

Comment:
Why are these changes needed? As far as I understand, this method is called even at HEAD from many places and this pull request doesn't alter the fact that this method returns null (my immediate instinct would be to move the function to get the dependency to CppSemantics, but I don't know the background)

Reply:
This change is to make `getGrepIncludes` always return `null` with Bazel. Bazel has a `grep-includes.sh` in `tools/cpp/grep-includes.sh` that unconditionally fails. It's harmless to pass that around as an `Artifact` until include scanning decides it wants to make a remote include scanning spawn, which is a possibility now for Bazel under this CL. If `getGrepIncludes` always returns `null`, include scanning will never attempt remote include scanning.

Moving to the `*Semantics` classes sounds fine to me if that's what you prefer. There are two process hiccups: 1) the CL size would go up due to modifying the "many" callsites you mention. 2) `GoogleCppSemantics.java` is outside my ability to edit, making the import process more complex.

The change here is admittedly not the nicest. Before modifying this method, I tried a simpler approach: removing `CcIncludeScanningRule`, which provides `grep-includes` as an implicit attribute, as an ancestor from all the Bazel cc rules. That worked elegantly for C++. Unfortunately the Objective-C rules use `CcIncludeScanningRule`, too, and do not appear to be forked between Blaze and Bazel.

Reply:
Yeah, this hack stuck out like a sore thumb when reviewing this change :)

I can imagine two possible approaches:
1. The more hacky (but still much nice than this one) approach is to give the constructor of `CcIncludeScanningRule` a Boolean flag that tells whether to add the extra attribute or not. This requires a single line of change in our internal-only code.
2. The more principled approach is moving the "use remote include scanning" bit to `CppSemantics`. The internal-only changes caused by this would be pretty straightforward but it's indeed a complication and I'd much rather to this refactoring atomically. 

So I guess (1) it is, unless @oquenchil thinks otherwise?

Reply:
I've now implemented approach 1, which is definitely less lines of hacks than messing with `getGrepIncludes`.

Reply:
Thanks. It looks much nicer :)